Assets/HoloToolkit/Sharing/Scripts/Utilities/ManualIpConfiguration.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            if (SharingStage.Instance != null)
+            {
+                SharingStage.Instance.SharingManagerConnected += OnConnected;
+                SharingStage.Instance.SharingManagerDisconnected += OnDisconnected;
+            }
+            else
+            {
+                Debug.LogError("Unable to subscribe to Sharing Stage!");
+            }
+        }
+
+        private void OnEnable()
+        {
+            if (!firstRun)
+            {
+                isTryingToConnect = true;
+                ConnectToSharingService();
+            }
+            else
+            {
+                firstRun = false;
+            }
+        }
+
+        private void Update()
+        {
+            if (timerRunning && timer - Time.time < 0)
+            {
+                if (isTryingToConnect)
+                {
+                    isTryingToConnect = false;
+                    OnDisconnected();
+
+                    PlayerPrefs.SetString("SharingServerIp", "Not Connected");
+                }
+            }
+        }
+
+        private void OnDestroy()
+        {
+            if (SharingStage.Instance != null)
+            {
+                SharingStage.Instance.SharingManagerConnected -= OnConnected;
+                SharingStage.Instance.SharingManagerDisconnected -= OnDisconnected;
+            }
+        }
+
+        private void CheckConnection()
+        {
+            // SharingStage should be valid at this point, but we may not be connected.
+            if (SharingStage.Instance.IsConnected)
+            {
+                OnConnected();
+            }
+            else if (!timerRunning)
+            {
+                timer = Time.time + Timeout;
+                timerRunning = true;
+            }
+        }
+
+        private void OnConnected(object sender = null, EventArgs e = null)
+        {
+            timerRunning = false;
+            connectionIndicator.color = Color.green;
+            ipAddress.text = SharingStage.Instance.Connection.GetRemoteAddress().ToString();
+
+            PlayerPrefs.SetString("SharingServerIp", ipAddress.text);
+
+            if (HideWhenConnected && isTryingToConnect)
+            {
+                StartCoroutine(Hide());
+            }
+
+            isTryingToConnect = false;
+        }
+
+        private void OnDisconnected(object sender = null, EventArgs e = null)
+        {
+            timerRunning = false;
+
+            if (!isTryingToConnect)
+            {
+                connectionIndicator.color = Color.red;
+                ipAddress.text = "Not Connected";
+            }
+        }
+
+        public void InputString(string input)
+        {
+            timerRunning = false;
+            isTryingToConnect = false;
+
+            if (ipAddress.text.Contains("Connected") || ipAddress.text.Contains("127.0.0.1"))
+            {
+                ipAddress.text = string.Empty;
+            }
+
+            if (ipAddress.text.Length < 15)
```

Comment:
To prevent string overflow. The window does not resize automatically. Besides there's only so many digits needed for a local ip address.